scripts/timelapse-colorizer-data/generate_data.py

```diff
@@ -139,107 +117,62 @@ def make_frames(grouped_frames, output_dir, dataset, scale: float):
             )
         )
 
-        # Save bounding box to JSON
-        bbox_json = {"data": np.ravel(bbox_data).tolist()}  # flatten to 2D
-        with open(outpath + "/bounds.json", "w") as f:
-            json.dump(bbox_json, f)
 
+def make_features(a: pd.DataFrame, features, writer: ColorizerDatasetWriter):
+    """
+    Generate the outlier, track, time, centroid, and feature data files.
+    """
+    # Collect array data from the dataframe for writing.
 
-def make_features(a, features, output_dir, dataset, scale: float):
-    nfeatures = len(features)
-    logging.info("Making features...")
-
-    outpath = os.path.join(output_dir, dataset)
-
-    # TODO check outlier and replace values with NaN or something!
-    logging.info("Writing outliers.json...")
     outliers = a["is_outlier"].to_numpy()
-    ojs = {"data": outliers.tolist(), "min": False, "max": True}
-    with open(outpath + "/outliers.json", "w") as f:
-        json.dump(ojs, f)
-
-    # Note these must be in same order as features and same row order as the dataframe.
-    logging.info("Writing track.json...")
-    tracks = a["track_id"].to_numpy()
-    trjs = {"data": tracks.tolist()}
-    with open(outpath + "/tracks.json", "w") as f:
-        json.dump(trjs, f)
-
-    logging.info("Writing times.json...")
-    times = a["index_sequence"].to_numpy()
-    tijs = {"data": times.tolist()}
-    with open(outpath + "/times.json", "w") as f:
-        json.dump(tijs, f)
-
-    logging.info("Writing centroids.json...")
+    tracks = a[TRACK_ID_COLUMN].to_numpy()
```

Comment:
why isn't "is_outlier" ( the line above this one)  a symbolic constant?  I'm wondering whether mapping the strings up above just adds noise to someone trying to quickly understand the code

Reply:
same for centroid_x and centroid_y

Reply:
I do think the symbolic names are a bit more clear than the original labels for some columns (ex: `label_img` vs. `OBJECT_ID_COLUMN`). I didn't initially include outliers because the EMT Migration project doesn't include the column, but I'll add them where they're used.

Reply:
The people preparing these scripts "should" be the ones who know what their data columns mean. It just felt like an extra level of indirection that one has to read.   But as long as it's consistent I think it's ok.